src/main/java/essentials/TaskList.java

```diff
@@ -0,0 +1,256 @@
+package essentials;
+
+import exceptions.JimmyException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    private final ArrayList<Task> taskList = new ArrayList<>();
+    private final Parser parser = new Parser();
+
+    /**
+     * Constructor for essentials.TaskList class.
+     *
+     * @param storage The storage object.
+     * @throws JimmyException If the file cannot be loaded.
+     */
+    public TaskList(Storage storage) throws JimmyException {
+        storage.loadFileContents(taskList);
+    }
+
+    /**
+     * Constructor for essentials.TaskList class. For testing purposes only.
+     */
+    public TaskList() {
+
+    }
+
+    /**
+     * Creates a new task and adds it to the task list.
+     *
+     * @param instruction The type of task to be created.
+     * @param details     The details of the task to be created.
+     * @throws JimmyException If the task type is invalid or the details are invalid.
+     */
+    public void createNewTask(String instruction, String details) throws JimmyException {
+        Task newTask = null;
+        try {
+            switch (instruction) {
+            case "todo":
+                newTask = createNewTodo(details);
+                break;
+            case "deadline":
+                try {
+                    newTask = createNewDeadline(details);
+                    break;
+                } catch (IllegalArgumentException e) {
+                    throw new JimmyException("Please enter a valid deadline.");
+                }
+            case "event":
+                try {
+                    newTask = createNewEvent(details);
+                    break;
+                } catch (IllegalArgumentException e) {
+                    throw new JimmyException("Please enter a valid start and end date.");
+                }
+            default:
+                throw new IllegalArgumentException();
+            }
+        } catch (IllegalArgumentException e) {
+            throw new JimmyException("Please enter a valid task type.");
+        } catch (DateTimeParseException e) {
+            throw new JimmyException("Error: Cannot parse the date written by the user. "
+                    + "Please use the following format: dd-MMM-yyyy (e.g. 19-01-2025)");
+        }
+
+        taskList.add(newTask);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(newTask);
+        System.out.println(generateListCounter() + "\n");
+    }
+
+    /**
+     * Creates a new todo task.
+     *
+     * @param details The details of the todo task.
+     * @return The new todo task.
+     */
+    private Todo createNewTodo(String details) {
+        return new Todo(details, false);
+    }
+
+    /**
+     * Creates a new deadline task.
+     *
```

Comment:
I like the method names and variable names which are very good and easy to understand